doc/changelog/04-tactics/14848-btermdn-discriminate-evar.rst

```diff
@@ -0,0 +1,7 @@
+- **Changed:**
+  Use discrimination nets for evar-containing goals in all
+  auto-like tactics. It essentially makes the behaviour of undiscriminated
+  databases to be the one of discriminated databases where all constants are
+  considered transparent. This may introduce very rare fringe incompatibilities.
```

Comment:
```suggestion
  Use discrimination nets for goals containing evars in all
  tactics similar to :tacn:`auto`. It essentially makes the behavior of undiscriminated
  hint databases the same as discriminated databases where all constants are
  considered transparent. This may be incompatible with previous behavior in very rare cases.
```

I don't understand the difference between discriminated and undiscriminated, nor do I know what a discrimination net is.  Will readers understand?  Is it different from the `discriminate` tactic?  Also, can you be more specific than "essentially" or drop that word?

Reply:
> Will readers understand?

The previous state of this documentation snippet is *both* badly written and completely wrong whatsoever. Even knowing the internals of the code made me wonder what the original author meant. This section is definitely *not* targeting casual users and the intended audience will understand it.

> discrimination net

It's a standard datastructure for this kind of algorithms. AFAIK it originated in Prolog.

> Is it different from the discriminate tactic?

Completely unrelated.

> essentially

No, saying it's the same would be lying but the last thing I want is to explain the current differences between both. Anyways, I still have a few tweaking PRs in the pipeline that would warrant a removal of this adverb, but in the meantime I'd rather keep it so that I don't get complaints that the phrasing is not literally true.